File: Server/schemas/auth/auth_user_out.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field
from utils.constants import MAX_LENGTH_NAME, MAX_LENGTH_EMAIL

class AuthUserOut(BaseModel):
    id_auth_user: int
    email: str = Field(..., max_length=MAX_LENGTH_EMAIL)
    id_user: int
    first_names: str = Field(..., max_length=MAX_LENGTH_NAME)
    last_names: str = Field(..., max_length=MAX_LENGTH_NAME)
    common_name: str = Field(..., max_length=MAX_LENGTH_NAME)
    full_name: str = Field(..., max_length=MAX_LENGTH_NAME)
    campus: str = Field(..., max_length=MAX_LENGTH_NAME)
    created_at: datetime
    updated_at: datetime
    user_type: str  
    id_student: Optional[int] = None
    rol_student: Optional[str] = None
    rut_student: Optional[str] = None
    campus_student: Optional[str] = None
    id_admin: Optional[int] = None

File: Client/src/modules/auth/types/auth_user_out.ts
export default interface AuthUserOut {
  id_auth_user: number
  email: string
  id_user: number
  first_names: string
  last_names: string
  common_name: string
  full_name: string
  campus: string
  created_at: string // datetime
  updated_at: string // datetime
  user_type: string
  id_student?: number
  rol_student?: string
  rut_student?: string
  campus_student?: string
  id_admin?: number
} 